### ISSUE TITLE
function to return compiled array dimensions.

### ISSUE THREAD

**Issue by ajpotts** [Bears-R-Us/arkouda#3383]
**Feature Description**
A function on the chapel side that returns the value/values of `max_array_dims` that were used at compilation time.

**Example Behavior**
If `max_array_dims = 3`, then acceptable outputs would be: `proc maxArrayDims(){ return 3;}` or `proc allowedArrayDims(){return [0, 1, 2, 3];}`.

**Describe Workarounds**
Right now developers can read the json, but this may not be available to users, or could be altered after compile and give the incorrect result.

**Additional context**
N/A

**Comment by jeremiah-corrado** [comment]
Good idea. I can include that in https://github.com/Bears-R-Us/arkouda/pull/3371 since I'm changing how the config file works there anyway.
